libs/frontend/feature-site-assets/src/components/SelectAssetModal.vue

```diff
@@ -0,0 +1,288 @@
+        class="asset-name-input"
+        :placeholder="t('assets.name')"
+        @update:modelValue="updateSearch"
+      />
+      <PSButton
+        class="select-button"
+        size="medium"
+        :text="t('build.select')"
+        :disabled="!selectedAsset"
+        @click="emitSelect"
+      />
+    </div>
+    <div class="asset-filter-row">
+      <PSMultiselect
+        v-if="contentTypes?.length"
+        :value="filter.content_type"
+        valueKey="key"
+        :options="contentTypes"
+        :clearable="true"
+        :caret="true"
+        :placeholder="t('assets.content_type')"
+        class="asset-content-type"
+        @select="updateContentType"
+      />
+    </div>
+    <div class="assets-wrap">
+      <div v-if="loading" class="assets-spinner-wrap">
+        <PSSpinner class="assets-spinner" :scale="2" color="#2a17d6" />
+      </div>
+      <div v-else-if="!siteAssets?.length" class="no-asset">
+        {{ t('build.no_matching_results') }}
+      </div>
+      <div
+        v-else
+        v-for="asset in siteAssets"
+        class="asset-item"
+        :class="{ 'asset-item-selected': selectedAsset?.id === asset.id }"
+        :title="asset.name"
+        @click="selectedAsset = asset"
+      >
+        <img class="asset-image" :src="asset.url" :alt="asset.name" />
+        <div class="asset-name">
+          {{ asset.name }}
+        </div>
+        <div v-if="selectedAsset?.id === asset.id" class="asset-selected-mark">
+          <Check class="check-mark" color="#0ceabf" />
+        </div>
+      </div>
+    </div>
+  </Modal>
+</template>
+
+<script lang="ts" setup>
+import { computed, inject, onMounted, ref, toRefs, watch } from 'vue'
+import { useI18n } from 'vue-i18n'
+import {
+  Check,
+  Modal,
+  PSButton,
+  PSInput,
+  PSMultiselect,
+  PSSpinner,
+} from '@pubstudio/frontend/ui-widgets'
+import { ISiteViewModel } from '@pubstudio/shared/type-api-platform-site'
+import { useSites } from '@pubstudio/frontend/feature-sites'
+import { store } from '@pubstudio/frontend/data-access-web-store'
+import {
+  AssetContentType,
+  SiteAssetState,
+  IListPlatformSiteAssetsRequest,
+  ISiteAssetViewModel,
+} from '@pubstudio/shared/type-api-platform-site-asset'
+import { debounce } from '@pubstudio/shared/util-debounce'
+import { ApiInjectionKey } from '@pubstudio/frontend/data-access-injection'
+import { PSApi } from '@pubstudio/frontend/util-api'
+import { useSiteAssetApi } from '@pubstudio/frontend/data-access-api'
+
+const { t } = useI18n()
+
+const props = withDefaults(
+  defineProps<{
+    show: boolean
+    initialSiteId?: string
+    contentTypes?: AssetContentType[]
+  }>(),
+  {
+    initialSiteId: undefined,
+    contentTypes: undefined,
+  },
+)
+const { show, initialSiteId } = toRefs(props)
+
+const emit = defineEmits<{
+  (e: 'cancel'): void
+  (e: 'select', asset: ISiteAssetViewModel): void
+}>()
+
+const rootApi = inject(ApiInjectionKey) as PSApi
+const api = useSiteAssetApi(rootApi)
```

Comment:
Hi @sampullman 

I didn't reuse the `useSiteAssets` composable here because asset menu is using the same composable as well. Using it to fetch site assets will cause the existing result in asset menu to be erased.

https://github.com/samatechtw/pubstudio-builder/assets/59676941/f8aed7c8-54de-4411-8b8b-a146bcbff49d